lib/operationArguments.ts

```diff
@@ -20,4 +22,30 @@ export interface OperationArguments {
    * The signal which can be used to abort requests.
    */
   abortSignal?: AbortSignal;
-}
\ No newline at end of file
+
+  /**
+   * Callback which fires upon upload progress. Only used in the browser.
+   */
+  onUploadProgress?: (progress: TransferProgressEvent) => void;
+
+  /**
+   * Callback which fires upon download progress. Only used in the browser.
+   */
+  onDownloadProgress?: (progress: TransferProgressEvent) => void;
+}
+
+/**
+ * Assigns the properties of a RequestOptions object to an OperationArguments object.
+ * @param args the arguments object to use in the OperatonArguments
+ * @param options the RequestOptions to apply
+ * @return an OperationArguments object
+ */
+export function createOperationArguments(args: { [parameterName: string]: any }, options?: RequestOptionsBase): OperationArguments {
+  return {
+    arguments: args,
+    customHeaders: options && options.customHeaders,
```

Comment:
may want to just check `if(options) {...}` and assign this stuff.. It's a small code size/possible(?) performance change.

Reply:
I'm optimistic that this function will reduce the code size a bit instead of increasing it by adding onUpload/Download progress property initializers inline in the generated code.